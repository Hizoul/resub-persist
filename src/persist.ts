import * as get from "lodash.get"
import * as isArray from "lodash.isarray"
import * as set from "lodash.set"
import makeObjectFromPropKeys from "./makeObjectFromPropKeys"
import { IPersistableStore } from "./type"

const persist = (storage: any, stores: IPersistableStore[]) => {
  return new Promise((resolve) => {
    let neededCallbacks = isArray(stores) ? stores.length : 0
    const callb = () => {
      neededCallbacks--
      if (neededCallbacks === 0) {
        resolve()
      }
    }
    for (const store of stores) {
      storage.setItem(`store.${store.name}`, JSON.stringify(makeObjectFromPropKeys(store)), callb)
    }
  })
}

export default persist
