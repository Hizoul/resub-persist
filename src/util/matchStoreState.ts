import { IPersistableStore } from "../type"
import makeObjectFromPropKeys from "./getStoreState"

/**
 * Shortcut for matching a store state against a snapshot in jest
 * @param store Store to snapshot
 * @param msg Name of the snapshot
 */
const matchStoreState = (store: IPersistableStore, msg?: string) => {
  expect(makeObjectFromPropKeys(store)).toMatchSnapshot(msg)
}

export default matchStoreState
