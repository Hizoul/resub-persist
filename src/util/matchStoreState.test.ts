import persist from "../persist"
import memForage from "../testUtil/miniMemForage"
import TestStore1 from "../testUtil/testStore"
import matchStoreState from "./matchStoreState"

test("Cover Persist Proeprty Test", () => {
  matchStoreState(TestStore1, "Initial State")
  TestStore1.toggleMyBool()
  TestStore1.setMyString("asdf")
  matchStoreState(TestStore1, "After Changes")
})
