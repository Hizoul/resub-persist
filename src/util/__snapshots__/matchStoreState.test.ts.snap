// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cover Persist Proeprty Test: After Changes 1`] = `
Object {
  "myBool": true,
  "myString": "asdf",
}
`;

exports[`Cover Persist Proeprty Test: Initial State 1`] = `
Object {
  "myBool": false,
  "myString": "",
}
`;
